[Mac EWS] Add a step to prune com.apple.coresymbolicationd caches.

bls1999 opened the pull request:
#### 2bee0df0004bc322fd3c1065e97aae68c8e4fe50
<pre>
[Mac EWS] Add a step to prune com.apple.coresymbolicationd caches.
<a href="https://bugs.webkit.org/show_bug.cgi?id=291979">https://bugs.webkit.org/show_bug.cgi?id=291979</a>
<a href="https://rdar.apple.com/149892791">rdar://149892791</a>

Reviewed by Brianna Fan.

EWS is experiencing an issue on some macOS runs where the size of the
coresymbolicationd cache fills the disk. To mitigate this, we&apos;d like to add a
step to prune that cache if it gets too large.

* Tools/CISupport/Shared/steps.py:
    (PruneCoreSymbolicationdCacheIfTooLarge): Move existing step from build-webkit-org/steps.py, as it&apos;s now shared with EWS.
* Tools/CISupport/build-webkit-org/steps.py:
    (PruneCoreSymbolicationdCacheIfTooLarge): Delete existing step, as it&apos;s been moved to Shared/steps.py.
* Tools/CISupport/ews-build/factories.py:
    (Factory.__init__): Add step for macOS to the default steps.
* Tools/CISupport/ews-build/factories_unittest.py:
    (TestExpectedBuildSteps): Update unit tests to expect the new step.

Canonical link: <a href="https://commits.webkit.org/294025@main">https://commits.webkit.org/294025@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d3bb622b3428d80aee7c869de82e622b4bfe0f45

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/100634 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/20286 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/10585 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/105771 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/51222 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/102675 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/20594 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/28760 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/76634 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/51222 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/103641 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/15806 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/90898 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/56989 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/15619 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/8905 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/50598 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/85531 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/8980 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/108126 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/27752 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/20384 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/85588 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/100192 "Passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/28115 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/87099 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/85129 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/29824 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/7552 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/21740 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/16370 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/27687 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/27498 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/30816 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/29056 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

briannafan reviewed (approved):
Can we move `PruneCoreSymbolicationdCacheIfTooLarge` to `Shared/steps.py` given both are exactly the same?

bls1999 committed:
[Mac EWS] Add a step to prune com.apple.coresymbolicationd caches.
https://bugs.webkit.org/show_bug.cgi?id=291979
rdar://149892791

Reviewed by Brianna Fan.

EWS is experiencing an issue on some macOS runs where the size of the
coresymbolicationd cache fills the disk. To mitigate this, we'd like to add a
step to prune that cache if it gets too large.

* Tools/CISupport/Shared/steps.py:
    (PruneCoreSymbolicationdCacheIfTooLarge): Move existing step from build-webkit-org/steps.py, as it's now shared with EWS.
* Tools/CISupport/build-webkit-org/steps.py:
    (PruneCoreSymbolicationdCacheIfTooLarge): Delete existing step, as it's been moved to Shared/steps.py.
* Tools/CISupport/ews-build/factories.py:
    (Factory.__init__): Add step for macOS to the default steps.
* Tools/CISupport/ews-build/factories_unittest.py:
    (TestExpectedBuildSteps): Update unit tests to expect the new step.

Canonical link: https://commits.webkit.org/294025@main
